### PR TITLE
user topic updater

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem

### CODE REVIEW

File: kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/UserLDAInterestsRepo.scala [hunk 1]
```diff
@@ -41,10 +41,12 @@ class UserLDATopicRepoImpl @Inject() (
...
 
-  def getByUser(userId: Id[User], version: ModelVersion[DenseLDA])(implicit session: RSession): Option[UserTopicMean] = {
-    (for { r <- rows if (r.userId === userId && r.version === version) } yield r.userTopicMean).firstOption
+  def getByUser(userId: Id[User], version: ModelVersion[DenseLDA])(implicit session: RSession): Option[UserLDAInterests] = {
+    (for { r <- rows if (r.userId === userId && r.version === version) } yield r).firstOption
   }
-  def getUpdateTime(userId: Id[User], version: ModelVersion[DenseLDA])(implicit session: RSession): Option[DateTime] = {
-    (for { r <- rows if (r.userId === userId && r.version === version) } yield r.updatedAt).firstOption
+
+  def getFeatureByUser(userId: Id[User], version: ModelVersion[DenseLDA])(implicit session: RSession): Option[UserTopicMean] = {
```
Comment (stkem) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/UserLDAInterestsRepo.scala]: `getTopicMeanByUser`?

File: kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/URILDATopicRepo.scala [hunk 2]
```diff
@@ -88,4 +92,17 @@ class URILDATopicRepoImpl @Inject() (
     SequenceNumber[NormalizedURI](sql.as[Long].first max 0L)
   }
 
+  def getUserTopicHistograms(userId: Id[User], version: ModelVersion[DenseLDA])(implicit session: RSession): Seq[(LDATopic, Int)] = {
+    import StaticQuery.interpolation
+
+    val query =
```
Comment (stkem) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/URILDATopicRepo.scala:100]: ok for testing things, but I think this query is going to be too expensive as is for prolonged use. Can you make a short comment in the code for future maintainer? (i.e. "if there is a performance problem look here"-kinda thing)

File: kifi-backend/modules/cortex/app/com/keepit/cortex/models/lda/LDAUserDbUpdater.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        log.info(s"committing with seq = ${keep.seq.value}")
+        commitRepo.save(commitInfo.withSeq(keep.seq.value).withUpdateTime(currentDateTime))
+      }
+    }
+  }
+
+  private def processUser(user: Id[User]): Unit = {
+    val model = db.readOnlyReplica { implicit s => userTopicRepo.getByUser(user, representer.version) }
+    if (shouldComputeFeature(model)) {
+      val topicCounts = db.readOnlyReplica { implicit s => uriTopicRepo.getUserTopicHistograms(user, representer.version) }
```
Comment (stkem) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/models/lda/LDAUserDbUpdater.scala:58]: I'm a bit skeptical, wether this is going to work as giving significant discrimination, but I suppose it's worth a try :-)
(And we did see good large scale differentiation when we looked at the data manually)

**Comment by stkem** [comment]
made a few small comments. lgtm overall! Sorry for the delay.

File: kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/URILDATopicRepo.scala (hunk 2 above)
Comment (yingjieMiao) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/URILDATopicRepo.scala:100]: maybe. for people like me, this took 22ms. for people like Danny, took 70ms. For the famous Jeff, it took 233ms. 

in the long run, we could try incremental updates.

File: kifi-backend/modules/cortex/app/com/keepit/cortex/models/lda/LDAUserDbUpdater.scala (hunk 3 above)
Comment (yingjieMiao) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/models/lda/LDAUserDbUpdater.scala:58]: we will see. current priority is to get all pipelines ready. We can play with the data a bit more later this week.

Comment (stkem) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/models/lda/LDAUserDbUpdater.scala:58]: Yes! Eyes on the prize :-)

**Comment by yingjieMiao** [comment]
@stkem 
addressed your comments. merging.

**Pull request merged**